File: PROT.cpp
#include <string>
#include <map>

using namespace std;

string rna2protein (string rna)
{
  if (rna.length() % 3 == 0)
  {
    std::map<std::string, std::string> type;
    type["UUU] = "F";
    type["UUC] = "F";
    type["UUA] = "L";
    type["UUG] = "L";
    type["UCU] = "S";
    type["UCC] = "S";
    type["UCA] = "S";
    type["UCG] = "S";
    type["UAU] = "Y";
    type["UAC] = "Y";
    type["UAA] = "";
    type["UAG] = "";
    type["UGU] = "C";
    type["UGC] = "C";
    type["UGA] = "";
    type["UGG] = "W";
    type["CUU] = "L";
    type["CUC] = "L";
    type["CUA] = "L";
    type["CUG] = "L";
    type["CCU] = "P";
    type["CCC] = "P";
    type["CCA] = "P";
    type["CCG] = "P";
    type["CAU] = "H";
    type["CAC] = "H";
    type["CAA] = "Q";
    type["CAG] = "Q";
    type["CGU] = "R";
    type["CGC] = "R";
    type["CGA] = "R";
    type["CGG] = "R";
    type["AUU] = "I";
    type["AUC] = "I";
    type["AUA] = "I";
    type["AUG] = "M";
    type["ACU] = "T";
    type["ACC] = "T";
    type["ACA] = "T";
    type["ACG] = "T";
    type["AAU] = "N";
    type["AAC] = "N";
    type["AAA] = "K";
    type["AAG] = "K";
    type["AGU] = "S";
    type["AGC] = "S";
    type["AGA] = "R";
    type["AGG] = "R";
    type["GUU] = "V";
    type["GUC] = "V";
    type["GUA] = "V";
    type["GUG] = "V";
    type["GCU] = "A";
    type["GCC] = "A";
    type["GCA] = "A";
    type["GCG] = "A";
    type["GAU] = "D";
    type["GAC] = "D";
    type["GAA] = "E";
    type["GAG] = "E";
    type["GGU] = "G";
    type["GGC] = "G";
    type["GGA] = "G";
    type["GGG] = "G";
    string protein = "";
    for (int i = 0; i<rna.length(); i = i+3)
    {
      string s = rna.substr(i, 3);
      if (type[s] != "")
      {
        protein = protein + type[s];
      }
      else
      {
        break;
      }
    }
    return protein;
  }
  return "";
}

void main()
{
  //string rna = "AUGGCCAUGGCGCCCAGAACUGAGAUCAAUAGUACCCGUAUUAACGGGUGA";
  string rna;
  cout << "Enter the RNA string" << endl;
  cin >> rna;
  string protein = rna2protein(rna);
  cout << protein << endl;
}

File: PROT.cs
using System;

public class Program
{
  public static void Main()
  {
    string mrna;
    Console.WriteLine("Enter the mRNA");
    mrna = Console.ReadLine();
    string protein = rna2protein(mrna);
    Console.WriteLine(protein);
  }
  public static string rna2protein (string s)
  {
    if (s.Length % 3 == 0)
    {
      int x = s.Length / 3;
      string result = "";
      Dictionary <string, string> map = new Dictionary<string, string>();
      map.Add ("UUU", "F");
      map.Add ("UUC", "F");
      map.Add ("UUA", "L");
      map.Add ("UUG", "L");
      map.Add ("UCU", "S");
      map.Add ("UCC", "S");
      map.Add ("UCA", "S");
      map.Add ("UCG", "S");
      map.Add ("UAU", "Y");
      map.Add ("UAC", "Y");
      map.Add ("UAA", "");
      map.Add ("UAG", "");
      map.Add ("UGU", "C");
      map.Add ("UGC", "C");
      map.Add ("UGA", "");
      map.Add ("UGG", "W");
      map.Add ("CUU", "L");
      map.Add ("CUC", "L");
      map.Add ("CUA", "L");
      map.Add ("CUG", "L");
      map.Add ("CCU", "P");
      map.Add ("CCC", "P");
      map.Add ("CCA", "P");
      map.Add ("CCG", "P");
      map.Add ("CAU", "H");
      map.Add ("CAC", "H");
      map.Add ("CAA", "Q");
      map.Add ("CAG", "Q"); 
      map.Add ("CGU", "R");
      map.Add ("CGC", "R");
      map.Add ("CGA", "R");
      map.Add ("CGG", "R");
      map.Add ("AUU", "I");
      map.Add ("AUC", "I");
      map.Add ("AUA", "I");
      map.Add ("AUG", "M");
      map.Add ("ACU", "T");
      map.Add ("ACC", "T");
      map.Add ("ACA", "T");
      map.Add ("ACG", "T");
      map.Add ("AAU", "N");
      map.Add ("AAC", "N");
      map.Add ("AAA", "K");
      map.Add ("AAG", "K");
      map.Add ("AGU", "S");
      map.Add ("AGC", "S");
      map.Add ("AGA", "R");
      map.Add ("AGG", "R");
      map.Add ("GUU", "V");
      map.Add ("GUC", "V");
      map.Add ("GUA", "V");
      map.Add ("GUG", "V");
      map.Add ("GCU", "A");
      map.Add ("GCC", "A");
      map.Add ("GCA", "A");
      map.Add ("GCG", "A");
      map.Add ("GAU", "D");
      map.Add ("GAC", "D");
      map.Add ("GAA", "E");
      map.Add ("GAG", "E");
      map.Add ("GGU", "G");
      map.Add ("GGC", "G");
      map.Add ("GGA", "G");
      map.Add ("GGG", "G");
      for (int i = 0; i<x; i++)
      {
        string cons = s.Substring(i*3, 3);
        result = result + map[cons];
      }
      return result;
    }
    return "";
  }
}
